admin/src/lib.rs

```diff
@@ -0,0 +1,323 @@
+				.map_err(|e| format!("Error loading dest metadata: {:?}", e))?
+				.ok_or_else(|| format!("Error opening dest metadata file"))?;
+
+			let dest_columns = dest_meta.columns;
+
+			if args.clear_dest && std::fs::metadata(&args.dest_path).is_ok() {
+				std::fs::remove_dir_all(&args.dest_path).map_err(|e| format!("Error removing dest dir: {:?}", e))?;
+			}
+			std::fs::create_dir_all(&args.dest_path).map_err(|e| format!("Error creating dest dir: {:?}", e))?;
+
+			let mut dest_options = Options::with_columns(&args.dest_path, dest_columns.len() as u8);
+			dest_options.columns = dest_columns;
+			dest_options.sync_wal = false;
+			dest_options.sync_data = false;
+
+			parity_db::migrate(&db_path, dest_options, !args.overwrite, &args.force_columns)
+				.map_err(|e| format!("Migration error: {:?}", e))?;
+
+			if args.overwrite && std::fs::metadata(&args.dest_path).is_ok() {
+				std::fs::remove_dir_all(&args.dest_path).map_err(|e| format!("Error removing dest dir: {:?}", e))?;
+			}
+		},
+		SubCommand::Check(check) => {
+			let db = parity_db::Db::open_read_only(&options)
+				.map_err(|e| format!("Invalid db: {:?}", e))?;
+			if !check.index_value {
+				// Note that we should use enum parameter instead.
+				return Err("Requires one of the following check flag: --index-value".to_string());
+			}
+			let check_param = parity_db::CheckParam::new(
+				check.column,
+				check.range_start,
+				check.range_end,
+				check.display,
+				check.display_value_max,
+			);
+			db.check_from_index(check_param)
+				.map_err(|e| format!("Check error: {:?}", e))?;
+		},
+		SubCommand::Flush(_flush) => {
+			let db = parity_db::Db::open(&options)
+				.map_err(|e| format!("Invalid db: {:?}", e))?;
+			while !db.all_empty_logs() {
+				std::thread::sleep(std::time::Duration::from_millis(300));
+			}
+		},
+		SubCommand::Stress(bench) => {
+
+			let args = bench.get_args();
+			// avoid deleting folders by mistake.
+			options.path.push("test_db_stress");
+			if options.path.exists() && !args.append {
+				std::fs::remove_dir_all(options.path.as_path())
+					.map_err(|e| format!("Error clearing stress db: {:?}", e))?;
+			}
+
+			use crate::bench::BenchDb;
+			let db = bench::BenchAdapter::with_options(&options);
+
+			crate::bench::run_internal(args, db);
+		},
+	}
+	Ok(())
+}
+
+/// Admin cli command for parity-db.
+#[derive(Debug, StructOpt)]
+pub struct Shared {
+	/// Specify db base path.
+	#[structopt(long, short = "d", value_name = "PATH", parse(from_os_str))]
+	pub base_path: Option<PathBuf>,
+
+	/// Do not sync file on each flush.
+	#[structopt(long)]
+	pub no_sync: bool,
+
+	/// Register stat from those admin operations.
+	#[structopt(long)]
+	pub with_stats: bool,
+
+	/// Indicate the number of column, when using
+	/// a new or temporary db, defaults to one.
+	#[structopt(long)]
+	pub nb_columns: Option<u8>,
+
+	/// Indicate a metadata configuration to use.
+	#[structopt(long)]
+	pub use_metadata: Option<PathBuf>,
```

Comment:
This only makes sense for the migration. All the commands that require an existing database should simply accept path.

Reply:
For migration, is there a case where we want to override the source metadata?
I will just remove it.

Reply:
I can't think of such case. Metadata should always match source tables.